File: apps/langgraph/db/schema/fileSpecification.ts
import { serial, text, timestamp, integer, jsonb, pgTable, index, unique } from 'drizzle-orm/pg-core';

export const fileSpecification = pgTable('file_specifications', {
  id: serial('id').primaryKey(),
  canonicalPath: text('canonical_path').notNull(),
  description: text('description').notNull(),
  filetype: text('filetype').notNull(),
  subtype: text('subtype').notNull(), 
  language: text('language'),
  createdAt: timestamp('created_at').defaultNow().notNull(),
  updatedAt: timestamp('updated_at').defaultNow().notNull(),
}, (table) => {
  return {
    filetypeIdx: index('file_specifications_filetype_idx').on(table.filetype),
    subtypeIdx: unique('file_specifications_subtype_idx').on(table.subtype),
    canonicalPathIdx: index('file_specifications_canonical_path_idx').on(table.canonicalPath),
  };
});

export type FileSpecification = typeof fileSpecification.$inferSelect;
export type NewFileSpecification = typeof fileSpecification.$inferInsert;

File: rails_app/app/models/file_specification.rb
# == Schema Information
#
# Table name: file_specifications
#
#  id             :integer          not null, primary key
#  canonical_path :string
#  description    :string
#  filetype       :string
#  component_type :string
#  language       :string
#  created_at     :datetime         not null
#  updated_at     :datetime         not null
#
# Indexes
#
#  index_file_specifications_on_canonical_path  (canonical_path)
#  index_file_specifications_on_component_type  (component_type)
#  index_file_specifications_on_filetype        (filetype)
#

class FileSpecification < ApplicationRecord
end
